ITE: soc: add cpu idle task

GTLin08 opened the pull request:
Implement the CPU idle task. The system should enter this task when
there is no any task to ensure power saving.

Tested on it8xxx2_evb board. It will reduce 12.5mA when system enters
the CPU idle task.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/39172)
<!-- Reviewable:end -->

brockus-zephyr reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/39172#-:-MlLnlKh3b0JvEAVnnwF:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dcpleung, @jackrosenthal, @keith-zephyr, @kgugala, @nashif, and @pgielda)



<!-- Sent from Reviewable.io -->

keith-zephyr reviewed (approved):
> Reviewed 2 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/39172)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dcpleung, @henrikbrixandersen, @jackrosenthal, @katsuster, @kgugala, @mgielda, @nashif, @pgielda, and @sjg20)



<!-- Sent from Reviewable.io -->

GTLin08 committed:
ITE: soc: add cpu idle task

Implement the CPU idle task. The system should enter this task when
there is no any task to ensure power saving.

Tested on it8xxx2_evb board. It will reduce 12.5mA when system enters
the CPU idle task.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>
